src/main/java/org/opensearch/knn/plugin/rest/RestSearchModelHandler.java

```diff
@@ -69,9 +70,22 @@ private void checkUnSupportedParamsExists(RestRequest request) {
         throw new IllegalArgumentException(errorMessage);
     }
 
+    private void searchSizeValidation(RestRequest request) {
+        ValidationException validationException = new ValidationException();
```

Comment:
I suggest we create instance of ValidationException when it's required, inside the nested if case.

Reply:
ack